cs/src/Contracts/TunnelAccessControlEntry.cs

```diff
@@ -79,6 +87,7 @@ public TunnelAccessControlEntry()
         /// For anonymous ACEs, this value is null.
         /// </remarks>
         [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        [StringLength(Providers.MaxLength)]
```

Comment:
Should we limit to known providers with a regular expression?

Reply:
Not worth it in my opinion - see my comment about a scopes regex.